### PR TITLE
Create a validation dataset for text models

### LINKED ISSUE #4: Create a squad like toy dataset with ROSA documentation
We would need a dataset to fine tune Large language models for searching ROSA documentation and evaluating the performance of different models. As a start, take existing FAQs and convert them into QA format. 
As a part of this issue, discuss and answer the following questions:

- [x] What Quick enough volume of QA data that we can manage? 100-1000 rows?
- [x] What kind of questions do we want the model to answer?
- [x] Do we want quick questions or longer ones or both?
- [x] Can the model point back to the resources?

### LINKED ISSUE #23: [POC] Create a notebook that generates ROSA QA dataset to evaluate the LLMs

### PR DESCRIPTION
**Pull request by Shreyanand**
Fixes #23

### CODE REVIEW

File: notebooks/create-validation-dataset.ipynb [hunk 1]
```diff
@@ -0,0 +1,225 @@
+{
```
Comment (codificat) [inline notebooks/create-validation-dataset.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.        if line.startswith('###'):
</pre><p> </p><p>A minor thing: we also want to skip lines that start with <code>## </code>  - the headings for the categories/groups. The current code seems to get these included as part of some answers to previous questions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/foundation-models-for-documentation/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (codificat) [inline notebooks/create-validation-dataset.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    validation_set
</pre><p> </p><p>Do we want to store the dataset somewhere?  Or maybe even just have the notebook write it to a file</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/foundation-models-for-documentation/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Comment by codificat** [comment]
Thanks for this!   I just added a couple of minor comments about the code.

> Fixes #4

It might also make #5 obsolete? Or at least partially - as it obtains basically the same data, probably in a cleaner form.

On the other hand, the obtained dataset does not include context, which is a problem for extractive QA. But I think that's fine if we focus on generative QA, where we'd also have to focus on other types of evaluation (beyond SQuAD).

File: notebooks/create-validation-dataset.ipynb (hunk 1 above)
Comment (suppathak) [reply notebooks/create-validation-dataset.ipynb:1]: <p>+1 </p><p>Shall we save it somewhere in the data folder? We will be using it in future for evaluating our generate QA model outputs. </p>

**Review by suppathak — COMMENTED** [review]
/lgtm

**Comment by Shreyanand** [comment]
> It might also make #5 obsolete? Or at least partially - as it obtains basically the same data, probably in a cleaner form.

We could focus that PR for creating a validation data for extractive methods.

**Comment by Shreyanand** [comment]
@codificat can we merge this if it looks good?

**Review by codificat — APPROVED** [review]
lgtm, thanks!

**Pull request merged**